Valet/VALValet.m

```diff
@@ -227,15 +227,25 @@ - (BOOL)isEqualToValet:(VALValet *)otherValet;
 
 - (BOOL)canAccessKeychain;
 {
-    NSString *canaryKey = @"VAL_KeychainCanaryUsername";
-    NSString *canaryValue = @"VAL_KeychainCanaryPassword";
+    BOOL canAccessKeychain = NO;
     
-    NSMutableDictionary *query = [self.baseQuery mutableCopy];
-    [query addEntriesFromDictionary:[self _secItemFormatDictionaryWithKey:canaryKey]];
-    [query addEntriesFromDictionary:@{ (__bridge id)kSecValueData : [canaryValue dataUsingEncoding:NSUTF8StringEncoding] }];
+    [self.lockForSetAndRemoveOperations lock];
+    {
+        NSString *const canaryKey = @"VAL_KeychainCanaryUsername";
+        NSString *const canaryValue = @"VAL_KeychainCanaryPassword";
+        
+        // Manually add the key to the keychain since we don't care about duplicates and are optimizing for speed.
+        NSMutableDictionary *query = [self.baseQuery mutableCopy];
+        [query addEntriesFromDictionary:[self _secItemFormatDictionaryWithKey:canaryKey]];
+        [query addEntriesFromDictionary:@{ (__bridge id)kSecValueData : [canaryValue dataUsingEncoding:NSUTF8StringEncoding] }];
```

Comment:
If we're adding one key/value, may be more efficient to do `-[setValue:forKey:]`